src/test/scala/org/broadinstitute/dsde/firecloud/integrationtest/ElasticSearchShareLogDAOSpec.scala

```diff
@@ -0,0 +1,107 @@
+package org.broadinstitute.dsde.firecloud.integrationtest
+
+import com.typesafe.scalalogging.LazyLogging
+import org.broadinstitute.dsde.firecloud.integrationtest.ESIntegrationSupport.{searchDAO, shareLogDAO}
+import org.broadinstitute.dsde.firecloud.model.ShareLog.{Share, ShareType}
+import org.scalatest.{BeforeAndAfterAll, FreeSpec, Matchers}
+
+import scala.util.Try
+
+class ElasticSearchShareLogDAOSpec extends FreeSpec with Matchers with BeforeAndAfterAll with LazyLogging {
+  override def beforeAll = {
+    // using the recreate from search dao because we don't have recreate in sharelog dao
+    searchDAO.recreateIndex()
+    ElasticSearchShareLogDAOSpecFixtures.fixtureShares map { share =>
+      shareLogDAO.logShare(share.userId, share.sharee, share.shareType)
+    }
+  }
+
+  override def afterAll = {
+    // using the delete from search dao because we don't have recreate in sharelog dao
+    searchDAO.deleteIndex()
+  }
+
+  "ElasticSearchShareLogDAO" - {
+    "logShare" - {
+      "should log a share and get it back successfully using the generated MD5 hash" in {
+        val share = Share("roger", "syd@gmail.com", ShareType.WORKSPACE)
+        val loggedShare = shareLogDAO.logShare(share.userId, share.sharee, share.shareType)
+        val check = shareLogDAO.getShare(share)
+        assertResult(loggedShare) { check }
+      }
+      "should successfully log a duplicate share" in {
+        val loggedShare = shareLogDAO.logShare("fake4", "fake3@gmail.com", ShareType.WORKSPACE)
+        val check = Try(shareLogDAO.logShare(loggedShare.userId, loggedShare.sharee, loggedShare.shareType))
+        assert(check.isSuccess)
+      }
+    }
+    "logShares" - {
+      "should log multiple shares for a single user and share type" in {
+        val userId = "fake2"
+        val shareType = ShareType.GROUP
+        val expected = ElasticSearchShareLogDAOSpecFixtures.fixtureShares.filter(_.userId == userId).filter(_.shareType == shareType)
+        val sharees = expected.map(_.sharee)
+        val check = shareLogDAO.logShares(userId, sharees, ShareType.GROUP)
+        assertResult(expected.map(s => (s.userId, s.sharee, s.shareType))) {
+          check.map(s => (s.userId, s.sharee, s.shareType))
+        }
+      }
+    }
```

Comment:
since everything in `ElasticSearchShareLogDAOSpecFixtures` is inserted into the index during `init`, this test is really only checking that `init` did its thing and that attempting to insert those rows again does not cause an error. Am I reading that right?

Reply:
that is right... seems a bit redundant now. i'll remove the `logShares` call and change the wording/logic to assert that `init` was successful.